src/file.cr

```diff
@@ -129,6 +129,14 @@ class File < IO::FileDescriptor
     seek value
   end
 
+  # Returns a File::Stat object for the named file or raises 
+  # an error based on LibC.stat(path, stat) return value (See File::Stat)
```

Comment:
I don't think that it calls `LibC.stat` under the hood is relevant to the end user (or shouldn't be normally at least). In any case its an implementation detail. How about some generic "or raises `Errno` in the case of an error"?